api/test/models/user.js

```diff
@@ -0,0 +1,41 @@
+import { expect, request } from 'chai';
+
+import User from '../../src/models/user';
+import FollowSchema from '../../src/models/follow';
+import LikeSchema from '../../src/models/like';
+import PinSchema from '../../src/models/pin';
+import PlaylistSchema from '../../src/models/playlist';
+import PodcastSchema from '../../src/models/podcast';
+import EpisodeSchema from '../../src/models/episode';
+import { loadFixture } from '../utils';
+
+describe('User model', () => {
+	let user;
+
+	before(async () => {
+		await loadFixture('user_model');
+		user = await User.findOne();
+		[PinSchema, FollowSchema, LikeSchema, PlaylistSchema].forEach(async (schema) => {
+			expect(await schema.find({user})).to.be.an('array').that.is.not.empty;
+		})
+	});
+
+	after(async () => {
+		[User, FollowSchema, LikeSchema, PinSchema, PlaylistSchema, PodcastSchema, EpisodeSchema].forEach(async (model) => {
+			await model.remove();
+		})
+	});
+
+	describe('remove document', () => {
+
+		it('should remove the User and all Pins, Playlists, Follows and Likes with a foreign key to the User', async () => {
+			await user.remove();
+			[PinSchema, FollowSchema, LikeSchema, PlaylistSchema].forEach(async (schema) => {
```

Comment:
We either need to await for the completion of futures that are created and discarded as result of running `forEach` here or replace this with a `for` loop so the `await`ing happens in the context of async function passed to `it`.
I'm not sure we are actually getting correct results for this test otherwise.

Reply:
This applies to `forEach` loops at line 18 in `before` and line 24 in `after`.